solutions/avnm-secured-hub-and-spoke/README.md

```diff
@@ -15,44 +15,45 @@ urlFragment: virtual-network-manager-secured-hub-and-spoke
 
 This sample deploys Azure virtual networks in a hub and spoke configuration, using Azure Virtual Network Manager to manage Virtual Network connectivity and implement sample Security Admin Rules. A VPN Gateway and test VMs are deployed to complete the hub and spoke features.
 
-
 ## Deploy sample
 
 ### Step 1: Create a Resource Group for the sample resources
 
 Create a resource group for the deployment.
 
-```azurecli-interactive
-az group create --name hub-spoke --location eastus
+```bash
+az group create --name rg-hub-spoke-eastus --location eastus
 ```
 
-### Step 2: Deploy infrastructure and Virtual Network Manager resources
+### Step 2: Clone repository and navigate to the correct folder
 
-```azurecli-interactive
-az deployment group create \
-    --resource-group hub-spoke \
-    --template-uri https://raw.githubusercontent.com/mspnp/samples/main/solutions/avnm-secured-hub-and-spoke/armTemplates/avnmResources.json
+```bash
+git clone https://github.com/mspnp/samples.git
+cd ./samples/solutions/avnm-secured-hub-and-spoke/bicep
 ```
 
-### Step 3: Deploy Virtual Network Manager Dynamic Network Group Policy resources
+### Step 3: Deploy infrastructure and Virtual Network Manager resources
+
+```bash
+# Generate ssh key and get public data.
+ssh-keygen -t rsa -b 2048
 
-```azurecli-interactive
-az deployment subscription create \
-    --template-uri https://raw.githubusercontent.com/mspnp/samples/main/solutions/avnm-secured-hub-and-spoke/armTemplates/avmnDynamicMembershipPolicy.json
+az deployment sub create --location eastus --template-file main.bicep -n avnm-secured-hub-and-spoke --parameters resourceGroupName=rg-hub-spoke-eastus sshKey="$(cat ~/.ssh/id_rsa.pub)"
 ```
 
 ## Solution deployment parameters
 
-| Parameter | Type | Description | Default |
-|---|---|---|--|
-| `location` | string | Deployment location | `resourceGroup().location` | 
-| `adminUserName` | string | The admin user name for deployed VMs. | `admin-avnm` |
-| `adminPassword` | securestring | The admin password for deployed VMs. | `null` |
+| Parameter       | Type         | Description                           | Default                    |
+| --------------- | ------------ | ------------------------------------- | -------------------------- |
+| `location`      | string       | Deployment location                   | `resourceGroup().location` |
```

Comment:
it seems the default is not the resource group location but the deployment.

If we decide to go with existent instead of creating, can we explore the idea of removing this param from bicep?

Reply:
Location was removed as parameter, the same for resource group name.